File: android/src/main/kotlin/com/vanvixi/file_saver_ffi/models/FileType.kt
package com.vanvixi.file_saver_ffi.models


data class FileType(
    var ext: String,
    var mimeType: String
) {

    init {
        ext = ext.lowercase()
        mimeType = mimeType.lowercase()
    }

    enum class Category {
        IMAGE,
        VIDEO,
        AUDIO,
        CUSTOM
    }

    val category: Category
        get() = when {
            mimeType.startsWith("image/") -> Category.IMAGE
            mimeType.startsWith("video/") -> Category.VIDEO
            mimeType.startsWith("audio/") -> Category.AUDIO
            else -> Category.CUSTOM
        }

    val isImage get() = category == Category.IMAGE
    val isVideo get() = category == Category.VIDEO
    val isAudio get() = category == Category.AUDIO
}

File: ios/Classes/FileSaver/Models/FileType.swift
import Foundation

struct FileType {
    let ext: String
    let mimeType: String

    enum Category {
        case image
        case video
        case audio
        case custom
    }

    var category: Category {
        if mimeType.starts(with: "image/") {
            return .image
        } else if mimeType.starts(with: "video/") {
            return .video
        } else if mimeType.starts(with: "audio/") {
            return .audio
        } else {
            return .custom
        }
    }
}
